packages/gatsby-image/src/index.js

```diff
@@ -238,25 +240,52 @@ const listenToIntersections = (el, cb) => {
   }
 }
 
-const noscriptImg = props => {
-  // Check if prop exists before adding each attribute to the string output below to prevent
-  // HTML validation issues caused by empty values like width="" and height=""
-  const src = props.src ? `src="${props.src}" ` : `src="" ` // required attribute
-  const sizes = props.sizes ? `sizes="${props.sizes}" ` : ``
-  const srcSet = props.srcSet ? `srcset="${props.srcSet}" ` : ``
-  const title = props.title ? `title="${props.title}" ` : ``
-  const alt = props.alt ? `alt="${props.alt}" ` : `alt="" ` // required attribute
-  const width = props.width ? `width="${props.width}" ` : ``
-  const height = props.height ? `height="${props.height}" ` : ``
-  const crossOrigin = props.crossOrigin
-    ? `crossorigin="${props.crossOrigin}" `
-    : ``
-  const loading = props.loading ? `loading="${props.loading}" ` : ``
-  const draggable = props.draggable ? `draggable="${props.draggable}" ` : ``
+const noscriptImg = (props) => {
+  const {
+    src = "",
+    alt = "",
+    sizes,
+    srcSet,
+    title,
+    width,
+    height,
+    crossOrigin,
+    loading,
+    draggable,
+  } = props
 
+  const imageProps = {
+    src,
+    alt,
+    sizes,
+    srcSet,
+    title,
+    width,
+    height,
+    crossOrigin,
+    loading,
+    draggable,
```

Comment:
not sure if this is prettier or this:
```suggestion
  const imageProps = {
    src: props.src || "",
    alt: props.alt || "",
    sizes: props.sizes,
    srcSet: props.srcSet,
    title: props.title,
    width: props.width,
    height: props.height,
    crossOrigin: props.crossOrigin,
    loading: props.loading,
    draggable: props.draggable,
```

Reply:
It's a bit better by compressing the two, as destructing only to build up an object again throws the reader off a bit, and as this list has been growing, it may just be better diff wise for changes with the compact version you're suggesting here.

An alternative via a utility method would allow for the best of both worlds I think, if we can justify the method with usage elsewhere:

```js
function withoutExcess(src, excess) {
  const trimmed = {};
    
  Object.keys(src)
    .filter(k => !excess[k])
    .forEach(k => { trimmed[k] = src[k]; });
  
  return trimmed;
};

// pluck the props to use
const {
    src,
    alt,
    sizes,
    srcSet,
    title,
    width,
    height,
    crossOrigin,
    loading,
    draggable,
    ...excess
  } = props

// Set defaults in new prop object you'll use them from, then spread the props object with excess props removed
const imageProps = {
    src = ``,
    alt = ``,
    ...withoutExcess(props, excess);
}
```

I think the method is pretty easy to read and usage is descriptive for the most part? With this approach, we avoid needlessly repeating ourselves, filtering out additional noise.